File: frontend/src/app/models/recipe.ts
/**
 * represents a virtual coach of the app
 */
export interface Coach {
    name: string;
    description: string;
    imageUrl: string;
}

/**
 * represents a generated recipe from the API
 */
export interface Recipe {
    dishName: string;
    dishDescription: string;
    ingredients: string;
    steps: string[];
    imageUrl: string;
    coach: Coach;
}

File: backend/data/recipe.py
import dataclasses
from typing import Optional

from .coach import Coach


@dataclasses.dataclass
class Recipe:
    dishName: str
    dishDescription: str
    ingredients: str
    steps: list[str]
    coach: Optional[Coach] = None

    def as_dict(self):
        return dataclasses.asdict(self)
